### DISCUSSION TITLE
Which version to build right now?

### DISCUSSION THREAD

**Discussion by chriseward** [sanni/cartreader#394] [General Chat]
It sounds like v5 still has some tweaks in progress. v3 is supposed to be more difficult to diy. v4 the way to go?

**Comment by sanni** [comment]
First the good news all the different versions can do the same stuff and will be fully supported in future firmware releases. So it's more about personal preference meaning you can't choose wrong.

## V3
The V3 is actually PCB revision 17, so it had 13 more revisions than the latest V5. That's why V3 seems more refined.
### Benefits:
- has been tested for 3 years and countless cartridges have been dumped with it. 
- once it's upgraded with the battery add-on it becomes super reliable and versatile for GB/GBA, N64, SNES and Megadrive. 
- feature complete, everything just works
### Drawbacks:
- Dumping NES can be challenging on the v3 because you need an adapter and then this "cartridge tower" sits very tall and unstable in the SNES slot.
- the buttons can break with time and replacing them is a little tricky 
- if it doesn't work, it is pretty hard to find out what part is responsible

## V4 (single slot build)
With V4(single slot build) I started completely from scratch not using any of the old V3 design files. It focuses on making building super easy, especially if you do the no-battery build option. 
### Benefits:
- every slot has its own PCB, so if a slot breaks you only need to replace that one slot not the whole cart reader
- easy to build new slot adapters for new systems
- the rotary encoder makes menu navigation so much easier than the buttons
- it's a very quick build, think 15 minutes instead of 2 hours as with the v3
- build-in battery support
### Drawbacks:
- People didn't like the chaos having 7 different cart slot adapter PCBs creates
- the pins of the cart slot adapters will bend from multiple insertions, so you will be constantly bending them back, which gets old pretty quickly
- uses the smaller Mega 2560 Pro where no official Arduino hardware is available and it's hard to get the correct schematics for the different variants.

## V4 (7-slot build)
The 7-slot v4 build was just an update for v4 owners so that they won't feel left out once v5 had released, I don't recommend it for new builds.

## V5
v5 (rev3) is the best build so far. It basically combines v3 and v4 together.
### Benefits:
- goes back to the beloved all-slots-on-one-PCB design meaning 7 slots are ready to go at all times
- main PCB can be replaced with a new version without too much hassle
- the rotary encoder makes menu navigation so much easier than the buttons
- compatible with V4 single slot adapters
### Drawbacks:
- needs a power bank or some other external power source
- missing some features right now like snesCIC support for SA1
- uses the smaller Mega 2560 Pro where no official Arduino hardware is available and it's hard to get the correct schematics for the different variants.

## Conclusion
Personally I would build the V5 (rev3) since the big screen and the rotary knob are a huge benefit compared to V3. I would not build the V4 since the V5 is just an updated V4. 
Maybe just start with the GBA slot on the V5 and see if you can get it to work and then advance to adding the six slot adapter, where you solder one slot at a time and test it before soldering the next slot. Then it should be as easy to build as the V4.

**Reply by chriseward** [reply]
Wow. Thank you so much for the thorough reply! Your advice about building the v5 in phases sounds like a good approach for me. Not too worried about lacking snesCIC support for SA1. I assume that will be added at some point.
